src/modules/keyboardmanager/ui/ShortcutControl.cpp

```diff
@@ -55,6 +55,237 @@ void ShortcutControl::AddNewShortcutControlRow(StackPanel& parent, Shortcut orig
+                    // If not, add a new drop down
+                    else
+                    {
+                        AddDropDown(parent, rowIndex, colIndex);
+                    }
+                }
+                // If last drop down and a modifier is selected but there are already 5 drop downs: warn the user
+                else if (IsModifierKey(keyCodeList[selectedKeyIndex]) && parent.Children().Size() >= 5)
+                {
+                    // warn and reset the drop down
+                    warningMessage.Text(L"Shortcuts must contain an action key");
+                    currentDropDown.ContextFlyout().ShowAttachedFlyout((FrameworkElement)currentDropDown);
+                    currentDropDown.SelectedIndex(-1);
+                }
+                // If None is selected but it's the last index: warn
+                else if (keyCodeList[selectedKeyIndex] == 0)
+                {
+                    // warn and reset the drop down
+                    warningMessage.Text(L"Shortcuts must contain an action key");
+                    currentDropDown.ContextFlyout().ShowAttachedFlyout((FrameworkElement)currentDropDown);
+                    currentDropDown.SelectedIndex(-1);
+                }
+                // If none of the above, then the action key will be set
+            }
+            // If it is the not the last drop down
+            else
+            {
+                if (IsModifierKey(keyCodeList[selectedKeyIndex]))
+                {
+                    // check if modifier has already been added before in a previous drop down
+                    std::vector<DWORD> currentKeys = GetKeysFromStackPanel(parent);
+                    bool matchPreviousModifier = false;
+                    for (int i = 0; i < currentKeys.size(); i++)
+                    {
+                        // Skip the current drop down
+                        if (i != dropDownIndex)
+                        {
+                            // If the key type for the newly added key matches any of the existing keys in the shortcut
+                            if (GetKeyType(keyCodeList[selectedKeyIndex]) == GetKeyType(currentKeys[i]))
+                            {
+                                matchPreviousModifier = true;
+                                break;
+                            }
+                        }
+                    }
+                    // If it matched any of the previous modifiers then reset that drop down
+                    if (matchPreviousModifier)
+                    {
+                        // warn and reset the drop down
+                        warningMessage.Text(L"Shortcut cannot contain a repeated modifier");
+                        currentDropDown.ContextFlyout().ShowAttachedFlyout((FrameworkElement)currentDropDown);
+                        currentDropDown.SelectedIndex(-1);
+                    }
+                    // If not, the modifier key will be set
+                }

```

Comment:
This check is already done before in the last dropdown case, it should probably be in a separate function